### PR TITLE
feat: add eugene migration linter for postgres

### PR DESCRIPTION
**Pull request by WhoIsSethDaniel**

### CODE REVIEW

File: lua/lint/linters/eugene.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    '--format=json',
+  },
+  stdin = false,
+  parser = function(output, _)
+    local diagnostics = {}
+    if #output > 0 then
+      local ok, decoded = pcall(vim.json.decode, output)
+      if not ok then
+        return diagnostics
+      end
```
Comment (mfussenegger) [inline lua/lint/linters/eugene.lua:14]:
```suggestion
      local decoded = vim.json.decode(output)
```

Decode errors would be caught further up in the framework and the error gets reported as diagnostic in the first line.

That can help catch configuration errors.

File: lua/lint/linters/eugene.lua [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+              source = 'eugene',
+              lnum = stmt.line_number,
+              col = 0,
+              severity = vim.diagnostic.severity.ERROR,
+              message = string.format('%s: %s: %s', tr.id, tr.name, tr.url),
+            })
+          end
+        end
+      end
+    end
```
Comment (mfussenegger) [inline lua/lint/linters/eugene.lua:28]:
```suggestion
        for _, tr in ipairs(stmt.triggered_rules) do
          table.insert(diagnostics, {
            source = 'eugene',
            lnum = stmt.line_number,
            col = 0,
            severity = vim.diagnostic.severity.ERROR,
            message = string.format('%s: %s: %s', tr.id, tr.name, tr.url),
          })
        end
      end
    end
```

The length check is kinda redundant

**Review by mfussenegger — COMMENTED** [review]
Could you also add it to the readme?

**Comment by WhoIsSethDaniel** [comment]
Added to README.
Applied suggested changes.

**Pull request merged**